src/olympia/devhub/views.py

```diff
@@ -586,13 +586,18 @@ def validate_addon(request):
 
 
 def handle_upload(filedata, request, channel, addon=None, is_standalone=False,
-                  submit=False):
+                  submit=False, source=amo.UPLOAD_SOURCE_DEVHUB):
     automated_signing = channel == amo.RELEASE_CHANNEL_UNLISTED
 
     user = request.user if request.user.is_authenticated else None
     upload = FileUpload.from_post(
         filedata, filedata.name, filedata.size,
-        automated_signing=automated_signing, addon=addon, user=user)
+        addon=addon,
+        automated_signing=automated_signing,
+        remote_addr=(core.get_remote_addr() or '')[:255],
```

Comment:
ip6 can't be more than 39 chars

Reply:
Technically, IPv4-mapped IPv6 could be 45 characters - and that's why we use that `max_length` value in `users`. I was worried here that we could get extra stuff because of proxies etc but that doesn't seem to be the case from our nginx configuration, so I'll change it to `45`.

Reply:
8398b3328a3909f67dc5f7f16f98969321b2c9bd and a97741a9e26f5680641c6f50e67f5fe53db75073